### ISSUE TITLE
g:switch_find_largest_match

### ISSUE THREAD

**Issue by maxfl** [AndrewRadev/switch.vim#38]
In short I propose to add a possibility to choose the longest match.

I have some common definitions for switching `yes`<->`no`.
For the latex there is a series of font size commands, which also include `\normalsize`. The latex configuration is added to the buffer configuration.

The problem is that when font size is switched to `\normalsize`, then the line is matched by `no` which is transformed to `yes`.

I would actually expect matching algorithm to choose the largest matching string for substitution. In this case it's guaranteed that the short match is not a substring of some longer match. Therefore I propose to add an option to check for the largest match. It may be also a string option, allowing to choose between several options: longest, shortest, sequential (or something like this).

**Comment by AndrewRadev** [comment]
The problem with yes/no is easily solveable by constraining the pattern:

``` vim
let g:switch_custom_definitions =
      \ [
      \   {
      \     '\<yes\>': 'no',
      \     '\<no\>':  'yes',
      \   },
      \ ]
```

This would only match "yes" and "no" when they're not parts of other words. Admittedly, it's more complicated than doing `["yes", "no"]`, which really makes me wonder if I made the right choice by not just adding the `\<`/`\>` in the first place. An alternative I've been thinking of for quite a while is doing something like `switch#WordList("yes", "no")` (which, it seems, already has a precedent, `switch#NormalizedCase`, I had forgotten about that).

For picking the larger string, I think I explained my reasoning in the documentation, but the gist of it is that, in many cases, pattern-matching simply wouldn't work if the plugin took the larger match. If you have an if-clause, for example:

``` ruby
if (...)
end
```

Anything within the brackets of the if-clause will be unswitchable, since the if-clause would the one that matches the "larger" switch. No true/false, no yes/no, no strings/symbols. This might not be an issue for your particular use cases, I don't know. Seems like one of those things that might surprise you, though.

I'll keep this in the pipeline, it shouldn't be *too* difficult to implement it, but consider the solution to your specific problem and see if you have any other issues.
